doc/guides/west/repo-tool.rst

```diff
@@ -1,79 +1,289 @@
+the manifest repository or as immediate subdirectories of the topdir.
+However, projects must have paths inside the workspace. (You may replace a
+project's repository directory within the workspace with a symbolic link to
+elsewhere on your computer, but west will not do this for you.)
+
+A workspace can contain additional Git repositories or other files and
+directories not managed by west. West basically ignores anything in the
+workspace except :file:`.west`, the manifest repository, and the projects
+specified in the manifest file.
+
+For upstream Zephyr, ``tinycbor`` and ``net-tools`` are projects. They are
+specified in the manifest file :file:`zephyr/west.yml`. This file specifies
+that ``tinycbor`` is located in the :file:`modules/lib/tinycbor` directory
+beneath the workspace topdir. By default, the Zephyr :ref:`build system
+<build_overview>` uses west to get the locations of all the projects in the
+workspace, so any code they contain can be used as :ref:`modules`.
+
+Finally, any repository managed by a west workspace (either the manifest
+repository or any project repository) can define :ref:`west-extensions`.
+Extensions are extra commands not built into west that you can run when
+using that workspace.
+
+The zephyr repository uses this feature to provide Zephyr-specific commands
+like :ref:`west build <west-building>`. Defining these as extensions keeps
+west's core agnostic to the specifics of any workspace's Zephyr version,
+etc.
+
+.. _west-struct:
+
+Structure
+*********
+
+West's code is distributed via PyPI in a Python package named ``west``.
+This distribution includes a launcher executable, which is also named
+``west`` (or ``west.exe`` on Windows).
+
+When west is installed, the launcher is placed by :file:`pip3` somewhere in
+the user's filesystem (exactly where depends on the operating system, but
+should be on the ``PATH`` :ref:`environment variable <env_vars>`). This
+launcher is the command-line entry point to running both built-in commmands
+like ``west init``, ``west update``, along with any extensions discovered
+in the workspace.
+
+In addition to its command-line interface, you can also use west's Python
+APIs directly. See :ref:`west-apis` for details.
+
+.. _west-manifest-rev:
+
+The ``manifest-rev`` branch
+***************************
+
+West creates and controls a Git branch named ``manifest-rev`` in each
+project. This branch points to the revision that the manifest file
+specified for the project at the time :ref:`west-update` was last run.
+Other workspace management commands may use ``manifest-rev`` as a reference
+point for the upstream revision as of this latest update. Among other
+purposes, the ``manifest-rev`` branch allows the manifest file to use SHAs
+as project revisions.
+
+Although ``manifest-rev`` is a normal Git branch, west will recreate and/or
+reset it on the next update. For this reason, it is normally a **bad idea**
```

Comment:
I would remove `normally`.
It should be considered a bad idea to do so.

Maybe there could be rare cases when it is needed, but it's still a bad idea.
```suggestion
reset it on the next update. For this reason, it is a **bad idea**
```

Reply:
How about we say `it is **dangerous**` instead?

I think there are legitimate, if advanced, use cases for manipulating it directly.

Reply:
**dangerous** sound good.

Reply:
Great, that's pushed.